src/core/thread/mle_router.cpp

```diff
@@ -3676,6 +3682,19 @@ otError MleRouter::RestoreChildren(void)
     }
 
 exit:
+
+    if ((index > kMaxChildren) || foundDuplicate)
+    {
+        // If there are more saved children in non-volatile settings
+        // than could be restored or there are duplicate entries with
+        // same extended address, we return OT_ERROR_FAILED to trigger
+        // a call to `RefreshStoredChildren()` to refresh the info and
+        // ensure that the non-volatile settings remain in sync with
+        // the child table.
+
+        error = OT_ERROR_FAILED;
```

Comment:
How about moving the `RefreshStoredChildren()` call to here?

Reply:
Do you mean removing `RefreshStoredChildren()`?

Reply:
I mean replacing this line with a call to `RefreshStoredChildren()` and make `RefreshStoredChildren()` a private method of `MleRouter`.

Reply:
Changed this in new pushed commit.